src/test/endian_conversion_test.cc

```diff
@@ -0,0 +1,179 @@
+      size_t current_byte_size = std::min(
+          partial_byte_size, (byte_size - partial_array_offset));
+      LittleEndianToHost(
+          datatype, partial_array, current_byte_size,
+          partial_result, next_offset);
+      partial_array += current_byte_size;
+    }
+
+    for (int i = 0; i < current_values.size(); i++) {
+      EXPECT_EQ(current_values[i], original_values[i]);
+    }
+  }
+}
+
+
+TEST(EndianConversionTest, ConvertBYTES)
+{
+  // BYTES array are prepended by size of content bytes
+  // That is the only portion which needs to switch byte order
+
+  const std::array<uint32_t, 10> original_values = {
+      0x00000004, 0xdeadbeef, 0x00000008, 0xdeadbeef, 0xdeadbeef,
+      0x0000000C, 0xdeadbeef, 0xdeadbeef, 0xdeadbeef, 0x00000000};
+
+
+  const std::array<uint32_t, 10> swapped_values = {
+      0x04000000, 0xdeadbeef, 0x08000000, 0xdeadbeef, 0xdeadbeef,
+      0x0C000000, 0xdeadbeef, 0xdeadbeef, 0xdeadbeef, 0x00000000};
+
+  ConversionTest(TRITONSERVER_TYPE_BYTES, original_values, swapped_values);
+}
+
+TEST(EndianConversionTest, ConvertUINT32)
+{
+  const std::array<uint32_t, 10> original_values = {
+      0xdeadbeef, 0xdeadbeef, 0xdeadbeef, 0xdeadbeef, 0xdeadbeef,
+      0xdeadbeef, 0xdeadbeef, 0xdeadbeef, 0xdeadbeef, 0xdeadbeef};
+
+  const std::array<uint32_t, 10> swapped_values = {
+      0xefbeadde, 0xefbeadde, 0xefbeadde, 0xefbeadde, 0xefbeadde,
+      0xefbeadde, 0xefbeadde, 0xefbeadde, 0xefbeadde, 0xefbeadde};
+
+  ConversionTest(TRITONSERVER_TYPE_UINT32, original_values, swapped_values);
+}
+
+TEST(EndianConversionTest, ConvertUINT16)
+{
+  const std::array<uint16_t, 10> original_values = {
+      0xdead, 0xdead, 0xdead, 0xdead, 0xdead,
+      0xdead, 0xdead, 0xdead, 0xdead, 0xdead};
+
+  const std::array<uint16_t, 10> swapped_values = {
+      0xadde, 0xadde, 0xadde, 0xadde, 0xadde,
+      0xadde, 0xadde, 0xadde, 0xadde, 0xadde};
+
+  ConversionTest(TRITONSERVER_TYPE_UINT16, original_values, swapped_values);
+}
+
+TEST(EndianConversionTest, ConvertUINT64)
+{
+  const std::array<uint64_t, 10> original_values = {
+      0xdeadbeefdeadbeef, 0xdeadbeefdeadbeef, 0xdeadbeefdeadbeef,
+      0xdeadbeefdeadbeef, 0xdeadbeefdeadbeef, 0xdeadbeefdeadbeef,
+      0xdeadbeefdeadbeef, 0xdeadbeefdeadbeef, 0xdeadbeefdeadbeef,
+      0xdeadbeefdeadbeef};
+
+  const std::array<uint64_t, 10> swapped_values = {
+      0xefbeaddeefbeadde, 0xefbeaddeefbeadde, 0xefbeaddeefbeadde,
+      0xefbeaddeefbeadde, 0xefbeaddeefbeadde, 0xefbeaddeefbeadde,
+      0xefbeaddeefbeadde, 0xefbeaddeefbeadde, 0xefbeaddeefbeadde,
+      0xefbeaddeefbeadde};
+
+  ConversionTest(TRITONSERVER_TYPE_UINT64, original_values, swapped_values);
+}
+
+TEST(EndianConversionTest, UINT8)
+{
+  const std::array<uint8_t, 10> original_values = {
+      0xde, 0xad, 0xde, 0xad, 0xde, 0xad, 0xde, 0xad, 0xde, 0xad};
+
+
+  ConversionTest(TRITONSERVER_TYPE_UINT8, original_values, original_values);
+}
+
+
+}}  // namespace triton::server
```

Comment:
I would like to see tests for floating pint as well to make sure the mantissa bits are read correctly on a big-endian machine.

Reply:
Looks like this is covered now.

Reply:
added tests for fp32 / fp64 -